src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -815,18 +816,14 @@ namespace ScatterKernelDetail{
 
 
       template < typename T_Obj>
-      static void* initHeap(const T_Obj& obj, void* pool, size_t memsize){
-        T_Obj* heap;
-        MALLOCMC_CUDA_CHECKED_CALL(cudaGetSymbolAddress((void**)&heap,obj));
+      static void* initHeap(T_Obj* heap, void* pool, size_t memsize){
```

Comment:
And actually `T_Obj&` should be `Scatter_Impl&`. It shouldn't be possible to pass another type in here that just happens to have a similar interface...